YoutubeMusicGeniusLyrics.user.js

```diff
@@ -286,19 +286,40 @@ function addLyrics (force, beLessSpecific) {
   genius.f.loadLyrics(force, beLessSpecific, songTitle, songArtistsArr, musicIsPlaying)
 }
 
+function getYoutubeMainVideo () {
+  let video = document.querySelector('#movie_player video[src]')
+  if (video !== null) {
+    return video
+  }
+  video = document.querySelector('video[src]')
+  if (video !== null) {
+    return video
+  }
+  return null
+}
+
 let lastPos = null
-function updateAutoScroll () {
+function updateAutoScroll (video, force) { // eslint-disable-line no-unused-vars
   let pos = null
-  try {
-    const [current, total] = document.querySelector('.ytmusic-player-bar .time-info.ytmusic-player-bar').textContent.split('/').map(s => s.trim()).map(s => s.split(':').reverse().map((d, i, a) => parseInt(d) * Math.pow(60, i)).reduce((a, c) => a + c, 0))
-    pos = current / total
-  } catch (e) {
-    // Could not parse current song position
-    pos = null
-  }
-  if (pos != null && !Number.isNaN(pos) && lastPos !== pos) {
-    genius.f.scrollLyrics(pos)
+  if (!video) {
+    video = getYoutubeMainVideo()
+  }
+  if (video) {
+    pos = video.currentTime / video.duration
+  }
+  if (pos !== null && pos >= 0 && `${lastPos}` !== `${pos}`) {
     lastPos = pos
+    const ct = video.currentTime
+    if (force === true) {
+      genius.f.scrollLyrics(pos)
+    } else {
+      setTimeout(() => {
+        const ct1 = video.currentTime
+        if (ct1 - ct < 50 / 1000 && ct1 > ct) {
+          genius.f.scrollLyrics(ct1 / video.duration)
+        }
+      }, 30)
```

Comment:
This does not work for me. The `if(...)` is false because ct1 and ct are always equal for me in Firefox.
Also I don't understand the purpose of it.

Reply:
Same problem in the Spotify script

Reply:
The purpose is that if the autoscroll shall be only for the media is playing.
Let's delete it first :)